Add IQS015 rule (#2379)
home-assistant/developers.home-assistant: docs/core/integration-quality-scale/rules/iqs015.md

---

> title: "IQS015 - Entities use DeviceInfo correctly"

Reviewer: ```suggestion
title: "IQS015 - Entities use DeviceInfo"
```

---

IQS015 is as follows:

> Entities use DeviceInfo correctly

Reviewer: ```suggestion
Entities use DeviceInfo
```

## Reasoning

> Devices are used to group entities into a single device.

Reviewer: I think we need to have one sentence that introduces the device registry in Home Assistant. Otherwise this sentence is confusing.

This is useful, since users usually think they add a device to their system, not a single entity.
In order for the user to have the best experience, the fields of the DeviceInfo should ideally be filled out.

## Example implementation

> In this example, there is a sensor entity that provides the device it will be added to.

Reviewer: ```suggestion
In this example, there is a sensor entity that defines which device it should be added to in the device registry, together with some metadata about the device.
```

---
title: "IQS015 - The integration creates devices"
---

IQS015 is as follows:

The integration creates devices

## Reasoning

> Devices are used to group entities to represent a single physical device.

Reviewer: ```suggestion
Devices, in Home Assistant, are used to group entities to represent either a single physical device or a service.
```

> This is useful, since users usually think they add a device to their system, not a single entity.

Reviewer: ```suggestion
This is useful, since users usually think they add a device or a service to their system, not a single entity.
```
